pkg/api/relationshipTuples.go

```diff
@@ -7,21 +7,28 @@ import (
 	"github.com/permitio/permit-golang/pkg/models"
 	"github.com/permitio/permit-golang/pkg/openapi"
 	"go.uber.org/zap"
+	"time"
 )
 
 type RelationshipTuples struct {
-	permitBaseApi
+	PermitBaseFactsApi
 }
 
 func NewRelationshipTuplesApi(client *openapi.APIClient, config *config.PermitConfig) *RelationshipTuples {
 	return &RelationshipTuples{
-		permitBaseApi{
-			client: client,
-			config: config,
-			logger: config.Logger,
+		PermitBaseFactsApi{
+			permitBaseApi{
+				client: client,
+				config: config,
+				logger: config.Logger,
+			},
 		},
 	}
 }
+func (u *RelationshipTuples) WaitForSync(timeout *time.Duration) *RelationshipTuples {
+	u.PermitBaseFactsApi.WaitForSync(timeout)
```

Comment:
PermitBaseFactsApi.WaitForSync returns a new client that you initialized inside the function, but you doesn't return it from this function ( relevant for all the functions ).
This means that whatever WaitForSync is doing is currently unused

Reply:
If I'm correct you should do return NewRelationshipTuplesApi(u.PermitBaseFactsApi.WaitForSync(timeout)) or something like this

Reply:
true. fixed